### PR TITLE
Node 12 deprecation todos

### PR DESCRIPTION
**Pull request by mrickard**
Signed-off-by: mrickard <maurice@mauricerickard.com>

## Proposed Release Notes
Converted `/dev/null` to `os.devNull` in 

- test/unit/collector/serverless.test.js
- test/unit/serverless/aws-lambda.test.js

and converted `fs.promises.rmdir` to `fs.promises.rm` in 

- test/unit/config/config-location.test.js
- test/versioned-external/checkout-external-tests.js

as part of Node 12 deprecation.

## Links
https://issues.newrelic.com/browse/NR-35263

## Details

TODOs in test code noted these changes for when Node 12 support was dropped. Recursive use of fs.rmdir was deprecated, and its replacement fs.rm was introduced in Node 14.

### CODE REVIEW

**Review by bizob2828 — APPROVED** [review]
LGTM but we can't land this until we actually drop node 12. So I will approve but will wait when we move from draft to ready for review

**Comment by mrickard** [comment]
Rebased against `main` and force-pushed. (I see that the old Node 12 versioned test is now gone.)

**Review by jordigh — APPROVED** [review]

File: test/versioned-external/checkout-external-tests.js
Comment (bizob2828) [inline test/versioned-external/checkout-external-tests.js:14]: I don't think this has been completely addressed.  @michaelgoin mentioned fs/promises not available in node 12.  not sure what he wanted to use in there.  If this is ok you forgot to remove the comments above

**Review by bizob2828 — CHANGES_REQUESTED** [review]

File: test/unit/config/config-location.test.js
Comment (mrickard) [inline test/unit/config/config-location.test.js:52]: @bizob2828 We seem to have a contradiction here. This TODO says that `rm` is preferred for Node versions newer than 12, with `rmdir` being deprecated. The other comment in versioned-external, however, says the opposite.

Comment (bizob2828) [reply test/unit/config/config-location.test.js:52]: Agreed.

Comment (mrickard) [reply test/unit/config/config-location.test.js:52]: https://nodejs.org/api/fs.html#fspromisesrmdirpath-options says 

`To get a behavior similar to the rm -rf Unix command, use [fsPromises.rm()](https://nodejs.org/api/fs.html#fspromisesrmpath-options) with options { recursive: true, force: true }.`

Comment (mrickard) [reply test/unit/config/config-location.test.js:52]: Those are the Node 18 docs, so...`rm` seems like it should be preferred.

File: test/versioned-external/checkout-external-tests.js
Comment (mrickard) [reply test/versioned-external/checkout-external-tests.js:14]: Resolving; I'll remove this comment, which is worded in a way that suggests the opposite of its intent.

File: test/unit/config/config-location.test.js
Comment (bizob2828) [reply test/unit/config/config-location.test.js:52]: yea just remove the comments, but also update the `require('fs').promises` to be `require('fs/promises')`

**Review by bizob2828 — APPROVED** [review]

**Review by jordigh — APPROVED** [review]

**Pull request merged**
